filter_plugins/cisco_nxos.py

```diff
@@ -0,0 +1,443 @@
+            object_name = line.removeprefix("interface ")
+            if not object_name.startswith("Ethernet") and not object_name.startswith("port-channel"):
+                # This allows things like management interfaces to pass through
+                delete_section = False;
+                new_config.append(line)
+                continue
+
+            # Check if this is a managed interface, if so, don't remove the section
+            if object_name in interfaces and interfaces[object_name].get("managed"):
+                delete_section = False
+                new_config.append(line)
+                continue
+
+            # Delete section
+            delete_section = True
+            continue
+
+        if line.startswith("vlan "):
+            if not found_vlanrange_line:
+                # Skip vlan range line
+                found_vlanrange_line = True
+                continue
+
+            # This is a VLAN config
+            object_name = int(line.removeprefix("vlan "))
+            if object_name in vlans:
+                if vlans[object_name].get("managed"):
+                    delete_section = False
+                    new_config.append(line)
+                    continue
+                else:
+                    delete_section = True
+                    continue
+            else:
+                delete_section = True
+                continue
+
+        if not delete_section:
+            new_config.append(line)
+
+    return new_config
+
+def generateVLANConfig(vlans):
+    config = []
+    for id,fields in vlans.items():
+        if fields.get("managed"):
+            continue
+
+        config.append("")  # Newline
+        config.append(f"vlan {str(id)}")
+        config.append(f"  name {fields['name']}")
+
+    return config
+
+def generateINTFConfig(interfaces):
+    """
+    This is the main method that generates the cisco NXOS config from the interface manifest
+    The output is ONLY the lines generated from the manifest, which needs to be combined with the existing running config
+
+    :param interfaces: The interfaces manifest
+    :type interfaces: dict
+    :return: The generated NXOS config
+    :rtype: list
+    """
+    def getLAGMembers(fields):
+        members = []
+        if "lag-members" in fields:
+            members += fields["lag-members"]
+
+        if "lacp-members-active" in fields:
+            members += fields["lacp-members-active"]
+
+        if "lacp-members-passive" in fields:
+            members += fields["lacp-members-passive"]
+
+        outlist = []
+        for member in members:
+            outlist.append(f"interface {member}")
+
+        return outlist
+
+    config_dict = {}
+
+    config = []
+    for intf,fields in interfaces.items():
+        intf_num = intf.split(" ")[-1]
+
+        # "fanout" field
+        if "fanout" in fields:
+            fanout_fields = fields["fanout"]
+            nxos_mode_str = ""
+            if fanout_fields["mode"] == "quad":
```

Comment:
This appears to want field `mode`, but all the `fanout` sections in `host_vars` have `type` instead:

```
interfaces:
  hundredGigE 1/1:
    fanout:
      type: "quad"
      speed: "25G"
```